Ginger/GingerCoreNET/Run/GingerRunner.cs

```diff
@@ -1423,44 +1423,50 @@ public void ProcessReturnValueForDriver(Act act)
-                {                     
-                    IV.ValueForDriver = act.ValueExpression.Calculate(IV.Value);
+                if (act.ValueExpression == null)
+                {
+                    act.ValueExpression = new ValueExpression(ProjEnvironment, CurrentBusinessFlow, DSList);
                 }
-                else
+                act.ValueExpression.DecryptFlag = true;
+                foreach (var IV in act.InputValues)
                 {
-                    IV.ValueForDriver = string.Empty;
+                    if (!string.IsNullOrEmpty(IV.Value))
+                    {
+                        IV.ValueForDriver = act.ValueExpression.Calculate(IV.Value);
+                    }
+                    else
+                    {
+                        IV.ValueForDriver = string.Empty;
+                    }
                 }
-            }
 
-            //Handle actions which needs VE processing like Tuxedo, we need to calculate the UD file values before execute, which is in different list not in ACT.Input list
-            List<ObservableList<ActInputValue>> list = act.GetInputValueListForVEProcessing();
-            if (list != null) // Will happen only if derived action implemented this function, since it needs processing for VEs
-            {
-                foreach (var subList in list)
+                //Handle actions which needs VE processing like Tuxedo, we need to calculate the UD file values before execute, which is in different list not in ACT.Input list
+                List<ObservableList<ActInputValue>> list = act.GetInputValueListForVEProcessing();
+                if (list != null) // Will happen only if derived action implemented this function, since it needs processing for VEs
                 {
-                    foreach (var IV in subList)
+                    foreach (var subList in list)
                     {
-                        if (!string.IsNullOrEmpty(IV.Value))
-                        {                            
-                            IV.ValueForDriver = act.ValueExpression.Calculate(IV.Value);
-                        }
-                        else
+                        foreach (var IV in subList)
                         {
-                            IV.ValueForDriver = string.Empty;
+                            if (!string.IsNullOrEmpty(IV.Value))
+                            {
+                                IV.ValueForDriver = act.ValueExpression.Calculate(IV.Value);
+                            }
+                            else
+                            {
+                                IV.ValueForDriver = string.Empty;
+                            }
                         }
                     }
                 }
+                act.ValueExpression.DecryptFlag = true;
+            }
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, "Exception occurred in ProcessInputValueForDriver : ", ex);
```

Comment:
The error message don't add anything, we know it's exception by adding ex object and we know it happen in ProcessInputValueForDriver from exception stackTrace which will be added to the log from ex object. Error message should be something like
Reporter.ToLog(eLogLevel.ERROR, string."Error occured while calculating the Action input VE's", ex);